src/Types.jl

```diff
@@ -128,16 +263,154 @@ function Base.print(io::IO, s::VersionSpec)
 end
 Base.show(io::IO, s::VersionSpec) = print(io, "VersionSpec(\"", s, "\")")
 
-Base.convert(::Type{VersionSet}, v::VersionNumber) = VersionSet(v, Base.nextpatch(v))
-Base.convert(::Type{VersionSet}, r::VersionRange{0,0}) = VersionSet()
-Base.convert(::Type{VersionSet}, r::VersionRange{m,1}) where {m} =
-    VersionSet(VersionNumber(r.lower.t...), VersionNumber(r.upper[1]+1))
-Base.convert(::Type{VersionSet}, r::VersionRange{m,2}) where {m} =
-    VersionSet(VersionNumber(r.lower.t...), VersionNumber(r.upper[1], r.upper[2]+1))
-Base.convert(::Type{VersionSet}, r::VersionRange{m,3}) where {m} =
-    VersionSet(VersionNumber(r.lower.t...), VersionNumber(r.upper[1], r.upper[2], r.upper[3]+1))
-Base.convert(::Type{VersionSet}, s::VersionSpec) = mapreduce(VersionSet, ∪, s.ranges)
-Base.convert(::Type{Available}, t::Tuple{SHA1,Dict{UUID,VersionSpec}}) = Available(t...)
+const Requires = Dict{UUID,VersionSpec}
+
+function merge_requires!(A::Requires, B::Requires)
+    for (pkg,vers) in B
+        A[pkg] = haskey(A,pkg) ? intersect(A[pkg],vers) : vers
+    end
+    return A
+end
+
+satisfies(pkg::UUID, ver::VersionNumber, reqs::Requires) =
+    !haskey(reqs, pkg) || in(ver, reqs[pkg])
+
+const DepsGraph = Dict{UUID,Dict{VersionNumber,Requires}}
```

Comment:
Is this typealias worth it? I find it easier to read the code when this is spelled out.

Reply:
It's used in quite a few places (around 20 including argument type declarations).
Mostly, however, I was planning to switch from a typealias to a true struct at some point, to include the `uuid_to_name` bit, since to me it makes sense to lump them together.